### ISSUE TITLE
AutoComplete Laggy on Windows

### ISSUE THREAD

**Issue by IanLikesDonuts** [texstudio-org/texstudio#700]
TeXstudio 2.12.16 (git 2.12.16)
Using Qt Version 5.12.1, compiled with Qt 5.12.1 R

Device: Windows 10 1903, i5-8350U, 32GB Ram, SSD. 

Issue: Autocompletion is very slow.

Repro: Open new empty document, type \infty. About one second lag after every character. 

Same with existing documents and references etc. 

Having used TexStudio previously, I do not recall these issues (on significantly slower machines). I suspect something with Qt and Windows.

**Comment by saribekyan** [comment]
**TL;DR:** The portable version worked for me.

Home machine: Windows 10 Pro; i7-8650U 1.9GHz, 8.00 GB RAM, SSD
Work machine: Windows 10 Pro, i7-3770 3.4GHz, 16GB RAM

TeXStudio version and reproduction: as in the original post, example: https://youtu.be/ViKgPSHRbfI

I have endured the same issue for some time and definitely know that it wasn't always like this. Today I decided to try various things and here's what I found out.

I realised that I do not get this problem on my work machine. I had an older version there (2.12.8). I installed a newer version on the work machine (2.12.16). I think the issue is there, but since it is a faster machine, at least it autocompletes faster than I type, but the lag is noticeable.

I installed the older version (2.12.6) on my personal machine and, to my surprise, I still have the lagging. I do not know how to explain it.

I found out however that the portable and new version does not have a lag on either of the machines. In all cases I have been using a freshly installed TeXStudio.

**Comment by timhoffm** [comment]
Even a "fresh" install reuses the settings of a previously installed version at `%APPDATA%\texstudio`. In particular therein is a folder `completion`, which contains files with completion word lists (*.cwl). Please check if the lag persists if you remove them.

The portable version does not access `%APPDATA%\texstudio`, which might be the reason that it's faster.

**Comment by saribekyan** [comment]
Thanks for the reply. I should have mentioned that before each of the installations, I erased the %APPDATA%\texstudio folder.

I copied my settings and profile (textstudio.ini and the profile file) from the non-portable version to portable one, so I can't tell the difference now except that it's faster.

**Comment by kevinhikaruevans** [comment]
FWIW, this seems to be fixed (for me, at least) when using the 3.0.0alpha2 release. I'm not quite sure what has changed between the past couple versions though.

**Comment by sunderme** [comment]
so, we can close this ?

**Comment by youneszeboudj** [comment]
removing %APPDATA%\texstudio\completion made texstudio faster for me. But then files get generated when using the editor again, a periodic cleaning should be planed. I am thinking of a corn task or a windows task until the issue gets fixed in the editor.
